client/signup/steps/website-content/index.tsx

```diff
@@ -39,13 +45,43 @@ function WebsiteContentStep( {
 	goToNextStep,
 	queryObject,
 }: WebsiteContentStepProps ) {
+	const [ formErrors, setFormErrors ] = useState< ValidationErrors >( {} );
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
 	const websiteContent = useSelector( getWebsiteContent );
 	const currentIndex = useSelector( getWebsiteContentDataCollectionIndex );
 	const siteId = useSelector( ( state ) => getSiteId( state, queryObject.siteSlug as string ) );
 	const siteCategory = useSelector( ( state ) => getDIFMLiteSiteCategory( state, siteId ) );
-	const dispatch = useDispatch();
-	const translate = useTranslate();
-	const [ formErrors, setFormErrors ] = useState< ValidationErrors >( {} );
+	const isWebsiteContentSubmitted = useSelector( ( state ) =>
+		isDIFMLiteWebsiteContentSubmitted( state, siteId )
+	);
+	const isSiteInformationLoaded = useSelector( ( state ) =>
+		isRequestingSite( state, siteId as number )
+	);
+
+	// We assume that difm lite is purchased when the is_difm_lite_in_progress sticker is active in a given blog
+	const isDifmLitePurchased = useSelector( ( state ) => isDIFMLiteInProgress( state, siteId ) );
+
+	//Make sure site information is loaded so that we can validate access to this page
+	useEffect( () => {
+		siteId && dispatch( requestSite( siteId ) );
+	}, [ siteId ] );
+
+	useEffect( () => {
+		if ( isSiteInformationLoaded && ! isDifmLitePurchased ) {
+			debug( 'DIFM not purchased yet, redirecting to DIFM purchase flow' );
+			page( `/start/do-it-for-me?siteSlug=${ queryObject.siteSlug }` );
```

Comment:
```suggestion
			page( `/start/do-it-for-me` );
```
The `do-it-for-me` flow does not accept `siteSlug` via query params, so we can remove this.